tests/unit/test_form.py

```diff
@@ -195,6 +195,28 @@ def test_build_formdata(self):
         expected = {'body': '<!-- @browser: None -->\n<!-- @ua_header: None -->\n<!-- @reported_with: None -->\n\n**URL**: http://\u611b\n\n**Browser / Version**: None\n**Operating System**: None\n**Tested Another Browser**: Unknown \n\n**Problem type**: Unknown\n**Description**: None\n**Steps to Reproduce**:\nNone\n\n\n\n_From [webcompat.com](https://webcompat.com/) with \u2764\ufe0f_', 'title': '\u611b - unknown'}  # noqa
         self.assertEqual(actual, expected)
 
+    def test_build_formdata_with_add_metadata(self):
+        """The data body sent to GitHub API, with additional metadata."""
+        # we just need to test that nothing breaks
+        # even if the data are empty
+        form_object = {'foo': 'bar'}
+        form_object = form.add_metadata(form_object, {'public_url':
+                                        'https://public.example.com'})
```

Comment:
so there might be an issue where it would be worth to test. Imagine the url is 

`https://public.example.com/life-->is-hard` or maybe even `https://public.example.com/life--is-hard`

`--` signifies the end of a comment. 

hmmm just tested and it seems that both webcompat and github manages things.